pkg/virt-handler/heartbeat/heartbeat_test.go

```diff
@@ -0,0 +1,89 @@
+package heartbeat
+
+import (
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/testutils"
+	clusterutil "kubevirt.io/kubevirt/pkg/util/cluster"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+var _ = Describe("Heartbeat", func() {
+
+	var node *v1.Node
+	var fakeClient *fake.Clientset
+
+	BeforeEach(func() {
+		node = &v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "mynode",
+			},
+		}
+		fakeClient = fake.NewSimpleClientset(node)
+	})
+
+	table.DescribeTable("with cpumanager featuregate should set the node to", func(deviceController device_manager.DeviceControllerInterface, schedulable string) {
+		heartbeat := NewHeartBeat(fakeClient.CoreV1(), deviceController, config(virtconfig.CPUManager), "mynode", clusterutil.ClusterTypeUnknown)
+		heartbeat.do()
+		node, err := fakeClient.CoreV1().Nodes().Get("mynode", metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(node.Labels).To(HaveKeyWithValue(virtv1.NodeSchedulable, schedulable))
+		Expect(node.Labels).To(HaveKeyWithValue(virtv1.CPUManager, "false"))
```

Comment:
Is it possible to fake  the cpuManagerPath  file content (or the isCPUManagerEnabled function) and also test for CPUManager=true ?

Reply:
definitely. Will add it tomorrow.

Reply:
Done.